README.md
---
layout: home
title: "jsFractalZoom-media - Media files for `jsFractalZoom`"
image: assets/favimage-840x472.jpg
---

\[click on image to start `zoomer` at shown location\]  
[![favimage](assets/favimage-840x472.jpg)](https://rockingship.github.io/jsFractalZoom?x=-0.8665722560433728&y=0.2308933773688535&r=3.021785750590329e-7&a=0&density=0.0362&iter=10080&theme=6&seed=2140484823)

# jsFractalZoom-media

Media files for `jsFractalZoom` (https://github.com/RockingShip/jsFractalZoom)

!!  
!! This repository contains about 800M of media files  
!!

### Welcome to the Wonderful World of (fractal) zooming and splash video codec

*When illustrations are 193x larger than the code*

This project contains the media files for: [https://github.com/RockingShip/jsFractalZoom](https://github.com/RockingShip/jsFractalZoom)

## Manifest

  - [gallery/index.html](https://rockingship.github.io/jsFractalZoom-media/gallery/index.html)  
    The fractal zoomer image gallery.

  - [videos/select-ade.html](https://rockingship.github.io/jsFractalZoom-media/videos/select-ade.html)  
    `splash` codec demo: "ADE" PixelsPerFrame selector (contains flashing white lights).

  - [videos/select-gta.html](https://rockingship.github.io/jsFractalZoom-media/videos/select-gta.html)  
    `splash` codec demo: "GTA" PixelsPerFrame selector (contains reckless driving).

  - [videos/ade-sbs-1800x506.webp](https://rockingship.github.io/jsFractalZoom-media/videos/ade-sbs-1800x506.webp)  
    [videos/ade-sbs-820x236.webp](https://rockingship.github.io/jsFractalZoom-media/videos/ade-sbs-820x236.webp)  
    Side-By-Side video of above for settings `PPF`=100 and upscaled 90x50.

  - [videos/gta-sbs-1800x506.webp](https://rockingship.github.io/jsFractalZoom-media/videos/gta-sbs-1800x506.webp)  
    [videos/gta-sbs-820x236.webp](https://rockingship.github.io/jsFractalZoom-media/videos/gta-sbs-820x236.webp)  
    Side-By-Side video of above for settings `PPF`=100 and upscaled 90x50.

  - [assets/ade-border-1920x1080.mp4](https://rockingship.github.io/jsFractalZoom-media/assets/ade-border-1920x1080.mp4)  
    [assets/ade-border-840x472.webp](https://rockingship.github.io/jsFractalZoom-media/assets/ade-border-840x472.webp)  
    Video displaying how `splash` constructs a single frame of "ADE".  
    The border displays processed scan row/columns.  
    The number in the lower left corner is progress (ranging 0-1).  

## Source code

This repository contains about 800M of media files.

Checkout the latest code:

```sh
  git clone https://github.com/RockingShip/jsFractalZoom-media.git
```

There are no releases due to their size.

## Versioning

Using [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/RockingShip/jsFractalZoom-media/tags).

## License

This project is licensed under the GNU General Public License v3 - see the [LICENSE.txt](LICENSE.txt) file for details.  
The gallery is MIT licensed.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

Media repositories have no releases due to their size.

```
2021-05-03 14:38:22 Changed: Gallery links to actual html endpoint.
2021-03-22 22:35:42 Changed: Item order in navbar.
2021-03-22 15:06:41 Changed: Modernized SEO headers.
2021-03-22 14:54:16 Added: `moonwalk` theme.
2021-03-22 14:41:28 Changed: renamed `jsFractalZoom.html` to `index.html`.
2021-03-12 23:53:18 Added: `???-sbs-1800x506.mp4`.
2021-03-12 20:03:21 Added: `index.html`.
2021-03-12 17:25:33 Added: `videos/'.
2021-03-12 15:26:15 Added: `gallery/`.
2021-03-01 20:57:45 Initial commit.
```


videos/gen-gta.sh
#!/bin/bash

if [ -z "$FFMPEG" ]; then FFMPEG=ffmpeg; fi

# extract 20 seconds from original video
if [ ! -f "gta-orig.720p.mp4" ]; then
  $FFMPEG -loglevel warning -stats -i cVGRDjlmMFk.mp4 -ss 8.2 -t 20 gta-orig.720p.mp4
fi

# unpack original into separate frames
rm -rf gta-frames
mkdir -p gta-frames
$FFMPEG -loglevel warning -stats -i gta-orig.720p.mp4 -vf scale=900:506 gta-frames/gta-%03d.png -y

for PPF in 1 2 5 10 20 50 100 150 200 300 600 900 1200 1600 2000 2400; do
	echo gta-${PPF}-900x506.webp
	$FFMPEG -loglevel warning -stats -r 30 -i gta-frames/gta-%03d.png -c:v splash -ppf $PPF -ppk 2 splash-gta-${PPF}-900x506.mkv -y
	$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-${PPF}-900x506.mkv -q:v 85 gta-${PPF}-900x506.webp -y
done

# Upscale 90x50
echo Upscale 90x50
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-1-900x506.mkv  -vf scale=90:50   -crf 18 gta-90x50.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=420:236 -crf 18 gta-90x50-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=420:236 -q:v 85 gta-90x50-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=900:506 -crf 18 gta-90x50-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=900:506 -q:v 85 gta-90x50-900x506.webp -y

# Upscale 285x160 to 420x236
echo Upscale 285x160
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-1-900x506.mkv  -vf scale=285:160 -crf 18 gta-285x160.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=420:236 -crf 18 gta-285x160-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=420:236 -q:v 85 gta-285x160-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=900:506 -crf 18 gta-285x160-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=900:506 -q:v 85 gta-285x160-900x506.webp -y

# scale PPF=100 to 420x236
echo PPF=100
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-100-900x506.mkv  -vf scale=420:236 -crf 18 gta-100-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-100-900x506.mkv  -vf scale=900:506 -crf 18 gta-100-900x506.mkv -y

# Create gta side-by-size
echo SBS
$FFMPEG -loglevel warning -stats -i gta-100-420x236.mkv -i gta-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 gta-sbs-820x236.webp -y
$FFMPEG -loglevel warning -stats -i gta-100-900x506.mkv -i gta-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 gta-sbs-1800x506.webp -y

# preview
$FFMPEG -loglevel warning -stats -i gta-frames/gta-001.png -q 0 gta-900x506.jpg -y
$FFMPEG -loglevel warning -stats -i gta-100-420x236.mkv -i gta-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 gta-sbs-820x236.jpg -y
$FFMPEG -loglevel warning -stats -i gta-100-900x506.mkv -i gta-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 gta-sbs-1800x506.jpg -y


videos/gen-ade.sh
#!/bin/bash

if [ -z "$FFMPEG" ]; then FFMPEG=ffmpeg; fi

# extract 20 seconds from original video
if [ ! -f "ade-orig.720p.mp4" ]; then
  $FFMPEG -loglevel warning -stats -i ade-20191018_025041.mp4 -t 10 -vf scale=1280:720 ade-orig.720p.mp4
fi

# unpack original into separate frames
rm -rf ade-frames
mkdir -p ade-frames
$FFMPEG -loglevel warning -stats -i ade-orig.720p.mp4 -vf scale=900:506 ade-frames/ade-%03d.png -y

for PPF in 1 2 5 10 20 50 100 150 200 300 600 900 1200 1600 2000 2400; do
	echo ade-${PPF}-900x506.webp
	$FFMPEG -loglevel warning -stats -r 30 -i ade-frames/ade-%03d.png -c:v splash -ppf $PPF -ppk 2 splash-ade-${PPF}-900x506.mkv -y
	$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-${PPF}-900x506.mkv -q:v 85 ade-${PPF}-900x506.webp -y
done

# Upscale 90x50
echo Upscale 90x50
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-1-900x506.mkv  -vf scale=90:50   -crf 18 ade-90x50.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=420:236 -crf 18 ade-90x50-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=420:236 -q:v 85 ade-90x50-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=900:506 -crf 18 ade-90x50-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=900:506 -q:v 85 ade-90x50-900x506.webp -y

# Upscale 285x160 to 420x236
echo Upscale 285x160
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-1-900x506.mkv  -vf scale=285:160 -crf 18 ade-285x160.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=420:236 -crf 18 ade-285x160-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=420:236 -q:v 85 ade-285x160-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=900:506 -crf 18 ade-285x160-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=900:506 -q:v 85 ade-285x160-900x506.webp -y

# scale PPF=100 to 420x236
echo PPF=100
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-100-900x506.mkv  -vf scale=420:236 -crf 18 ade-100-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-100-900x506.mkv  -vf scale=900:506 -crf 18 ade-100-900x506.mkv -y

# Create ade side-by-size
echo SBS
$FFMPEG -loglevel warning -stats -i ade-100-420x236.mkv -i ade-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 ade-sbs-820x236.webp -y
$FFMPEG -loglevel warning -stats -i ade-100-900x506.mkv -i ade-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 ade-sbs-1800x506.webp -y

# preview
$FFMPEG -loglevel warning -stats -i ade-frames/ade-001.png -q 0 ade-900x506.jpg -y
$FFMPEG -loglevel warning -stats -i ade-100-420x236.mkv -i ade-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 ade-sbs-820x236.jpg -y
$FFMPEG -loglevel warning -stats -i ade-100-900x506.mkv -i ade-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 ade-sbs-1800x506.jpg -y

#
# $FFMPEG -loglevel warning -stats -r:v 240/1 -i ade-border-1920x1080.mp4 -vf scale=840:472 -r 12/1 ade-border-840x472.webp
